packages/@react-aria/utils/src/scrollIntoView.ts

```diff
@@ -63,3 +70,64 @@ function relativeOffset(ancestor: HTMLElement, child: HTMLElement, axis: 'left'|
   }
   return sum;
 }
+
+
+export function scrollIntoViewFully(target: Element, scrollOptions?: ScrollIntoViewOptions) {
+  if (scrollOptions == null) {
+    scrollOptions = {
+      block: 'nearest'
+    };
+  }
+
+  if (shouldScrollIntoView(target)) {
+    scrollIntoViewHelper(target, scrollOptions);
+  }
+}
+
+// Determines if a target element is partially/fully out of view and thus should be scrolled into view if possible.
+// Recurses through every scrollable parent of the target and returns true the target is not visible in any of the scrollable parents.
+function shouldScrollIntoView(target: Element, originalTarget?: Element) {
```

Comment:
Some problematic cases that I ran into:
- returning shouldScrollIntoView=true for partially out of view elements. Table row is always partially out of view if it is wider than the body
- table resizer/absolutely positioned elements. The resizer doesn't have a scrollable parent other than the root but can still be scrolled out of view since it is absolutely positioned and the table body is the scrollable element rather than the table header. If I switch the logic to look at each parentElement rather than the scrollParents, the resizer is always partially out of view of its parent element